tests/run/unicodemethods.pyx

```diff
@@ -26,16 +26,12 @@ def print_all(l):
     "//PythonCapiCallNode")
 def split(unicode s):
     """
-    >>> print_all( text.split() )
-    ab
-    jd
-    sdflk
-    as
-    sa
-    sadas
-    asdas
-    fsdf
-    >>> print_all( split(text) )
+    >>> def test_split():
+    ...     py = text.split()
+    ...     cy = split(text)
+    ...     assert py == cy, (py, cy)
+    ...     return cy
+    >>> print_all( test_split() )
```

Comment:
Instead of returning something, I think we can just move the call to `print_all()` into the test helper function as well.

The same applies to the `print()` calls in other tests. I think the only reason why we call `print()` explicitly is the 'u' prefix of unicode strings in Py2, which prevents a straight doctest output comparison.

Reply:
@scoder, thanks for feedback. I believe it is better to keep `print*` outside. Let me explain: `test_X` function verifies method X for consistency in between Py and Cy implementation and allows the test to verify the actual result. Some results are strings, some boolean, some integers. We use `print`/`print_all` only for strings and list of strings kinds of result. We do not use print for boolean and integer results.

So for consistency I suggest to keep all `test_*` functions as close as possible to each other: retrieve py and cy values, assert that they are the same and return that result for inspection. Then in the tests themselves it will be always clear what to expect from `test_X`: it does call `unicode.X` and returns the result.

Do you think that makes sense?